File: energy_hydrogen.f90
program energy_hydrogen
  implicit none
  double precision, external :: e_loc, psi
  double precision :: x(50), w, delta, energy, dx, r(3), a(6), norm
  integer :: i, k, l, j

  a = (/ 0.1d0, 0.2d0, 0.5d0, 1.d0, 1.5d0, 2.d0 /)

  dx = 10.d0/(size(x)-1)
  do i=1,size(x)
     x(i) = -5.d0 + (i-1)*dx
  end do

  do j=1,size(a)

     ! TODO

     print *, 'a = ', a(j), '    E = ', energy
  end do

end program energy_hydrogen

program energy_hydrogen
  implicit none
  double precision, external :: e_loc, psi
  double precision :: x(50), w, delta, energy, dx, r(3), a(6), norm
  integer          :: i, k, l, j

  a = (/ 0.1d0, 0.2d0, 0.5d0, 1.d0, 1.5d0, 2.d0 /)

  dx = 10.d0/(size(x)-1)
  do i=1,size(x)
     x(i) = -5.d0 + (i-1)*dx
  end do

  delta = dx**3

  r(:) = 0.d0

  do j=1,size(a)
     energy = 0.d0
     norm   = 0.d0
     
     do i=1,size(x)
        r(1) = x(i)

        do k=1,size(x)
           r(2) = x(k)

           do l=1,size(x)
              r(3) = x(l)

              w = psi(a(j),r)
              w = w * w * delta

              energy = energy + w * e_loc(a(j), r)
              norm   = norm   + w 
           end do

        end do

     end do

     energy = energy / norm
     print *, 'a = ', a(j), '    E = ', energy
  end do

end program energy_hydrogen

File: energy_hydrogen.c
#include "hydrogen.h"

int main() {
      const double a[6] = {0.1, 0.2, 0.5, 1., 1.5, 2.};
      const int sizex = 50;
      double energy, r[3];
      double x[sizex], dx, w, wsum, dV;

      dx = 10.0 / (sizex - 1);
      for (int i = 0; i < sizex; ++i) {
	    x[i] = -5.0 + (i - 1) * dx;
      }
      dV = dx * dx * dx;

      for (int l = 0; l < 6; ++l) {
	    energy = 0.0;
	    wsum = 0.0;
	    for (int i = 0; i < sizex; ++i) {
		  r[0] = x[i];
		  for (int j = 0; j < sizex; ++j) {
			r[1] = x[j];
			for (int k = 0; k < sizex; ++k) {
			      r[2] = x[k];
			      w = psi(a[l], r, 3) * psi(a[l], r, 3) * dV;
			      wsum += w;
			      energy += w * e_loc(a[l], r, 3);
			}
		  }
	    }
	    printf("a = %.2f  E = %.5f \n", a[l], energy / wsum); 
      }

      
      return 0;
}
